test/javascripts/acceptance/run-query-test.js.es6

```diff
@@ -0,0 +1,193 @@
+            data_type: "boolean",
+          },
+          {
+            column_name: "created_at",
+            data_type: "timestamp",
+          },
+          {
+            column_name: "updated_at",
+            data_type: "timestamp",
+          },
+        ],
+      });
+    });
+
+    server.get("/admin/plugins/explorer/queries", () => {
+      return helper.response({
+        queries: [
+          {
+            id: -6,
+            sql:
+              "-- [params]\n-- int :months_ago = 1\n\nWITH query_period AS (\n    SELECT\n        date_trunc('month', CURRENT_DATE) - INTERVAL ':months_ago months' as period_start,\n        date_trunc('month', CURRENT_DATE) - INTERVAL ':months_ago months' + INTERVAL '1 month' - INTERVAL '1 second' as period_end\n        )\n\n    SELECT\n        ua.user_id,\n        count(1) AS like_count\n    FROM user_actions ua\n    INNER JOIN query_period qp\n    ON ua.created_at >= qp.period_start\n    AND ua.created_at <= qp.period_end\n    WHERE ua.action_type = 1\n    GROUP BY ua.user_id\n    ORDER BY like_count DESC\n    LIMIT 100\n",
+            name: "Top 100 Likers",
+            description:
+              "returns the top 100 likers for a given monthly period ordered by like_count. It accepts a ‘months_ago’ parameter, defaults to 1 to give results for the last calendar month.",
+            param_info: [
+              {
+                identifier: "months_ago",
+                type: "int",
+                default: "1",
+                nullable: false,
+              },
+            ],
+            created_at: "2021-02-02T12:21:11.449Z",
+            username: "system",
+            group_ids: [],
+            last_run_at: "2021-02-11T08:29:59.337Z",
+            hidden: false,
+            user_id: -1,
+          },
+        ],
+      });
+    });
+
+    server.post("/admin/plugins/explorer/queries/-6/run", () => {
+      return helper.response({
+        success: true,
+        errors: [],
+        duration: 27.5,
+        result_count: 2,
+        params: { months_ago: "1" },
+        columns: ["user_id", "like_count"],
+        default_limit: 1000,
+        relations: {
+          user: [
+            {
+              id: -2,
+              username: "discobot",
+              name: null,
+              avatar_template: "/user_avatar/localhost/discobot/{size}/2_2.png",
+            },
+            {
+              id: 2,
+              username: "andrey1",
+              name: null,
+              avatar_template:
+                "/letter_avatar_proxy/v4/letter/a/c0e974/{size}.png",
+            },
+          ],
+        },
+        colrender: {
+          0: "user",
+        },
+        rows: [
+          [-2, 2],
+          [2, 2],
+        ],
+      });
+    });
+  });
+
+  test("it runs query and renders data and a chart", async function (assert) {
+    await visit("admin/plugins/explorer?id=-6");
+
+    assert.ok(
+      queryAll("div.name h1").text().trim() === "Top 100 Likers",
```

Comment:
Do we need `queryAll` when we're looking for only 1 element?

Reply:
Not really 😊. In most cases here, `query()` is enough. I've fixed this.